src/app/timed-call/timed-call.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'timed-call',
+  templateUrl: './timed-call.component.html',
+  styleUrls: ['./timed-call.component.scss']
+})
+export class TimedCallComponent implements OnInit, OnDestroy {
+
+  // Function that will be called everytime the Timer reaches 0.
+  @Input()
+  functionToCall: Function;
+
+  // Optional parameter to define the interval times that are needed in the parent Component. This overwrites the default of 10, 30, 60, 120 and 300 seconds if provided
+  @Input()
+  delayOptions?: number[];
+
+  // Optional parameter to provide another label for the Timer Component than "Refresh Rate"
+  @Input()
+  label?: string;
+
+  public callInterval: any;
+  public currentCallDelay: number;
+  public defaultDelayOptions: number[] = [10, 30, 60, 120, 300];
+  public circleVisible: boolean = true; 
+
+  constructor() { }  
+
+
+  ngOnInit(): void {
+    // Establish timeout interval, get the named reference for it to update it later
+    this.label = this.label ? this.label : "Refresh Rate"
+    this.delayOptions = (this.delayOptions ? this.delayOptions : this.defaultDelayOptions)
+    this.currentCallDelay = this.delayOptions[0];
+    this.callInterval = setInterval(() => {
+      this.functionToCall();
+     } , this.currentCallDelay * 1000);
+  }
+
+
+
+  changeDelay() {
+    //Set next value of the interval array as call delay
+    var index = this.delayOptions.indexOf(this.currentCallDelay);  
+    if (index < this.delayOptions.length - 1) {
+      this. currentCallDelay = this.delayOptions[index + 1];
+    } else {
+      this.currentCallDelay = this.delayOptions[0];
+    }        
```

Comment:
It would be nice to have the delay options cycle logic separated from the actual interval mangement logic. The cycle logic could be encapsulated like so:

```ts
delayOptions = cycle([10, 30, 60, 120, 300]);

// ...

       this.currentCallDelay = this.delayOptions.next();

// ...

function* cycle<T>(array: T[]) {
  while (true) yield* steps;
}
```

Of course, this requires some familiarity with JS generators.